### DISCUSSION TITLE
Something wrong with the routes?

### DISCUSSION THREAD

**Discussion by lsfratel** [mako-framework/framework#312] [General]
I think there is something wrong with the routes?
I have a route that makes 2 queries in the database and returns this data, but it is taking 360ms, the latency from my house to my server is 180ms, so I put a cache in this route and it is still taking 360ms, so I put a slash (/) at the end and realized that the framework made a 301 redirect to the route without the slash, and the return this time took 700ms, double the 360ms, the other routes are normal.

```php

# The route
$routes->group(['prefix' => 'v1'], function (Routes $routes): void
{
	...

	/*
	 * Mangas endpoints
	 */
	$routes->group(['prefix' => 'mangas', 'middleware' => ['cache']], function (Routes $routes): void
	{
		$routes->get('', [MangaController::class, 'list']);
		$routes->post('/create', [MangaController::class, 'create']);
		$routes->get('/count', [MangaController::class, 'count']);
		$routes->get('/{id}', [MangaController::class, 'show']);
	});

	...

});

# Function in controller
public function list(): JSON
{
	$qs = $this->request->getQuery();

	if ($qs->has('q'))
	{
		$mangas = (new Manga)->scope('search', $qs->get('q'))->limit(100)->all();
	}
	else
	{
		$mangas = (new Manga)->scope('latest')->limit(21)->all();
	}

	return $this->jsonResponse($mangas);
}

# ORM Scopes
public function searchScope($qs, $term): void
{
	$qs->select(['id', 'name', 'poster', 'description'])
		->whereRaw('MATCH(`name`) AGAINST (? IN NATURAL LANGUAGE MODE)', [$term])
		->including(['genres']);
}

public function latestScope($qs): void
{
	$qs->select(['id', 'name', 'poster', 'description'])
		->descending('updated_at')
		->including(['genres']);
}

# The cache
public function execute(Request $request, Response $response, Closure $next): Response
{
	if (getenv('ENV') == 'development')
	{
		return $next($request, $response);
	}

	$key = $request->getPath() . '?' . http_build_query($request->getQuery()->all());

	if ($this->cache->has($key))
	{
		return $response->setBody($this->cache->get($key))->setType('application/json');
	}

	$resp = $next($request, $response);

	$this->cache->put($key, $resp->getBody(), 60 * $this->minutes);

	return $resp;
}

# The database queries
"mysql": [
	{
		"query": "SELECT `id`, `name`, `poster`, `description` FROM `mangas` ORDER BY `updated_at` DESC LIMIT 21",
		"time": 0.0002510547637939453
	},
	{
		"query": "SELECT `genres`.*, `genres_mangas`.`manga_id` FROM `genres` INNER JOIN `genres_mangas` ON `genres_mangas`.`genre_id` = `genres`.`id` WHERE `genres_mangas`.`manga_id` IN (31750, 31749, 31748, 31747, 31746, 31745, 31683, 31672, 31633, 31608, 31586, 31394, 31314, 31313, 31231, 31193, 31009, 30984, 30908, 30800, 30797)",
		"time": 0.000514984130859375
	}
]
```
I thought it was the database queries, but the logs show that they are fast.
Even with caching enabled or without caching, the return time is the same.

So, Any ideas?

**Comment by freost** [comment]
That's weird. I tried to set up an empty project and tested a route with and without the trailing slash and there some minor overhead but a router redirect shouldn't take several hundred ms since it never hits any of the controllers or middleware. Could it be that the app establishes a database connection before a route action is executed?

<img width="1352" alt="Screenshot 2023-01-25 at 21 49 02" src="https://user-images.githubusercontent.com/773958/214687849-8df9b309-e75d-43ba-8f3b-a2492eecb17c.png">
<img width="1355" alt="Screenshot 2023-01-25 at 21 51 48" src="https://user-images.githubusercontent.com/773958/214687863-998b863a-296a-4d97-8985-5cc125a9dc7d.png">

**Reply by lsfratel** [reply]
No, all interactions with the database are in the controllers.

**Reply by lsfratel** [reply]
![Captura de tela de 2023-01-25 18-20-10](https://user-images.githubusercontent.com/43170875/214706274-3da6e1db-087b-4e3b-9cd0-985436b5096e.png)

**Comment by lsfratel** [comment]
The funny thing is that it is only on this route, the other routes are normal, this route does m2m queries in the database, all the keys have index, I don't know what is going on.
